main_test.go

```diff
@@ -55,10 +55,7 @@ func TestGofmt(t *testing.T) {
 	}
 }
 
-var testBackends = map[string]int{
-	"python2":  1,
```

Comment:
the reasonning was to always test for/with python2 and python2+cffi as python2 is the reference (for now)

admittedly, this makes only sense when testing with travis-CI.
perhaps we should initialize `testBackends` as you do now but test (inside `init()`) if we are running with travis and forcefully enable these 2 backends back again.

Reply:
I think we can do it the opposite way. Instead of detecting travis/user, we should by default attempt to run only when all the backends are available. Otherwise, the user has to explicitly state that he/she wants to run only on the available backends using an environment variable something like GOPY_TEST_SKIP_UNAVAILABLE_BACKENDS=1. Please see next patch.

Reply:
the problem with this is we may fail (in Travis) to detect that we are not testing _e.g._ py3 because for some reason the backend isn't detected.
detecting which backends are available and run those is fine for a user.

but for travis, I think we need to be explicit and fail hard if any given backend is failing.

Reply:
so I'd be in favor of modifying the `.travis.yml` file to export/define an environment variable (say `GOPY_TRAVIS_CI=1`), detect it inside `main_test.go` and act accordingly.

what do you think?

Reply:
Done.